### PR TITLE
[Core] Better implementation of OpenPosition.

### PR DESCRIPTION
**Pull request by makarid**
I deleted a class with the same name but not so useful.

### CODE REVIEW

File: xchange-kraken/src/main/java/org/knowm/xchange/kraken/KrakenAdapters.java [hunk 1]
```diff
@@ -85,20 +82,16 @@ public static OpenPositions adaptOpenPositions(
...
-                          .getCost()
-                          .divide(
-                              krakenOpenPosition
-                                  .getVolume()
-                                  .subtract(krakenOpenPosition.getVolumeClosed()),
-                              RoundingMode.HALF_EVEN),
-                      krakenOpenPosition.getFee()));
+                  new OpenPosition.Builder()
+                          .instrument(new CurrencyPair(krakenOpenPosition.getAssetPair()))
+                          .type(krakenOpenPosition.getType().equals(KrakenType.BUY)
```
Comment (earce) [inline xchange-kraken/src/main/java/org/knowm/xchange/kraken/KrakenAdapters.java:87]: enums should be ==

File: xchange-ftx/src/main/java/org/knowm/xchange/ftx/FtxAdapters.java [hunk 2]
```diff
@@ -280,12 +278,17 @@ public static OpenPositions adaptOpenPositions(List<FtxPositionDto> ftxPositionD
...
-                      new CurrencyPair(ftxPositionDto.getFuture()))
-                  .originalAmount(ftxPositionDto.getOpenSize())
-                  .build());
+          if(ftxPositionDto.getSize().compareTo(BigDecimal.ZERO) > 0){
+            openPositionList.add(
+                    new OpenPosition.Builder()
+                            .instrument(new CurrencyPair(ftxPositionDto.getFuture()))
+                            .price(ftxPositionDto.getEntryPrice())
+                            .size(ftxPositionDto.getSize())
+                            .type(ftxPositionDto.getSide().equals(FtxOrderSide.buy)
```
Comment (earce) [inline xchange-ftx/src/main/java/org/knowm/xchange/ftx/FtxAdapters.java:287]: enums should be ==

**Comment by earce** [comment]
How many classes use `OpenPosition.java`? I'm surprised you aren't seeing compilation issues with classes needing the classpath updated for that class?

**Comment by makarid** [comment]
it was only used from 2 classes KrakenAdapter and FtxAdapter.

Στις Τρί, 26 Ιαν 2021 στις 2:50 π.μ., ο/η Erick Arce <
notifications@github.com> έγραψε:

**Pull request merged**